ecs-cli/modules/clients/aws/cloudformation/template.go

```diff
@@ -113,7 +113,7 @@ var template = `
       "ConstraintDescription": "must be a valid EC2 instance type."
     },
     "KeyName": {
-      "Type": "AWS::EC2::KeyPair::KeyName",
+      "Type": "String",
```

Comment:
Why the change in type?

Reply:
This is the same strategy the Console uses in their cloud formation template as well. The `AWS::EC2::KeyPair::KeyName` type apparently goes through special [validation](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#AWS-Specific Parameter Types), which means the condition for checking if this value is an empty string won't work. Changing it to a string allows us to us the cloudformation conditions to set Keyname on the LaunchConfiguration

Reply:
Got it 👍